java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -298,7 +305,7 @@ public Description matchCompilationUnit(
     if (!initialized && !disabled) {
       try {
         init(flags);
```

Comment:
just noticing this even though not part of this change. Why was this under `matchCompilationUnit`? Ideally, `init` at the beginning right after construction would be better. Is this to handle the silent exception problem in the constructor?

Reply:
This is exactly to avoid the silent swallowing of the error inside the constructor and use of the zero-arg constructor. I think it is safer and easier to debug this way, but it is a bit of a workaround for the EP's "feature" of silent fallback of constructors :/ 

It's also because of this that I couldn't make the `ImmutableMultimap` also `final`. But I do think it's the best we can do right now.